src/cli/mod.rs

```diff
@@ -161,65 +264,95 @@ fn get_store<F: LurkField + for<'a> serde::de::Deserialize<'a>>(
 }
 
 macro_rules! new_repl {
-    ( $cli: expr, $field: path ) => {{
-        let limit = $cli.limit.unwrap_or(DEFAULT_LIMIT);
-        // let rc = $cli.rc.unwrap_or(DEFAULT_RC);
+    ( $cli: expr, $limit: expr, $rc: expr, $field: path, $backend: expr ) => {{
         let mut store = get_store(&$cli.zstore).with_context(|| "reading store from file")?;
         let env = store.nil();
-        // Repl::<$field, Coproc<$field>>::new(store, env, limit, rc)?
-        Repl::<$field, Coproc<$field>>::new(store, env, limit, DEFAULT_RC)?
+        Repl::<$field>::new(store, env, $limit, $rc, $backend)?
     }};
 }
 
 impl ReplCli {
     pub fn run(&self) -> Result<()> {
         macro_rules! repl {
-            ( $field: path ) => {{
-                let mut repl = new_repl!(self, $field);
+            ( $limit: expr, $rc: expr, $field: path, $backend: expr ) => {{
+                let mut repl = new_repl!(self, $limit, $rc, $field, $backend);
                 if let Some(lurk_file) = &self.load {
                     repl.load_file(lurk_file)?;
                 }
                 repl.start()
             }};
         }
-        match get_field()? {
-            LanguageField::Pallas => repl!(pallas::Scalar),
-            LanguageField::Vesta => repl!(vesta::Scalar),
-            LanguageField::BLS12_381 => repl!(blstrs::Scalar),
+        let config = get_config(&self.config)?;
+        let limit = get_parsed_usize("limit", &self.limit, &config, DEFAULT_LIMIT)?;
+        let rc = get_parsed_usize("rc", &self.rc, &config, DEFAULT_RC)?;
+        let field = get_parsed("field", &self.field, &config, parse_field, DEFAULT_FIELD)?;
+        let backend = get_parsed(
+            "backend",
+            &self.backend,
+            &config,
+            parse_backend,
+            DEFAULT_BACKEND,
+        )?;
+        match field {
+            LanguageField::Pallas => repl!(limit, rc, pallas::Scalar, backend),
+            // LanguageField::Vesta => repl!(limit, rc, vesta::Scalar, backend),
+            // LanguageField::BLS12_381 => repl!(limit, rc, blstrs::Scalar, backend),
+            LanguageField::Vesta => todo!(),
+            LanguageField::BLS12_381 => todo!(),
         }
```

Comment:
What if the field is fundamentally incompatible with the separatedly-parsed, yet not-validated backend?